src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -2568,13 +2538,17 @@ namespace OpenMS
       if (source == layers_view_)
       {
         // only the selected row can be dragged => the source layer is the selected layer
-        LayerData& layer = getActiveCanvas()->getCurrentLayer();
+        LayerDataBase& layer = getActiveCanvas()->getCurrentLayer();
 
         // attach feature, consensus and peak data
         FeatureMapSharedPtrType features = layer.getFeatureMap();
         ExperimentSharedPtrType peaks = layer.getPeakDataMuteable();
         ConsensusMapSharedPtrType consensus = layer.getConsensusMap();
-        vector<PeptideIdentification> peptides = layer.peptides;
+        vector<PeptideIdentification> peptides;
```

Comment:
maybe add some comment like
```suggestion
        // if the layer provides identification data -> retrieve it
        vector<PeptideIdentification> peptides;
```

Reply:
this code will go away soon, but I can add the comment in the meantime :)